microsoft/PowerToys
# [PT Run] Replace tblimp-Microsoft.Search.Interop package with source implementation

## Summary of the Pull Request
![image](https://user-images.githubusercontent.com/4016293/153976944-19aee31b-f2f2-4181-83a0-f984ef3f35f3.png)

**What is this about:**
This PR removes the tlbimp-Microsoft.Search.Interop NuGet package and replaces with COM Interop interfaces/classes to provide an implementation that works with what is currently used .

**What is included in the PR:** 
* tblimp-Microsoft.Search.Interop NuGet package removed
* SearchAPI implementation of SearchManager, SearchQueryHelper, and SearchCatalogManager
* Microsoft.Search.Interop.dll removed from Setup

**How does someone test / validate:** 
* Open Power Toys Run and enter a search term that should return results from files on your computer.
* Place a breakpoint at the end of the `ExecuteQuery` method in `WindowsSearchAPI.cs`, Set the`PowerLauncher` project as the startup project and debug it. Verify the `results` list contains results from your local search index.

## Quality Checklist

- [X] **Linked issue:** #8557

## Contributor License Agreement (CLA)
A CLA must be signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/PowerToys) and sign the CLA.
